### PR TITLE
refactor!: move drag and drop logic inside the Tree component

### PR DESCRIPTION
**Pull request by Kenny806**

### CODE REVIEW

File: src/components/Tree/Node.tsx [hunk 1]
```diff
@@ -173,15 +174,16 @@ export const TreeNode = ({
                     className="tw-p-0 tw-m-0 tw-font-sans tw-font-normal tw-list-none tw-text-left"
                     data-test-id="sub-tree"
                 >
-                    {renderNodeArray(nodes, activeNodeId, onClick, onDrop, [...parentIds, id])}
+                    {renderNodeArray(nodes, activeNodeId, listId, onClick, onDrop, [...parentIds, id])}
```
Comment (GoranRibic) [inline src/components/Tree/Node.tsx:177]: args for renderNodeArray are out of control - we should make it more compact

File: src/components/Tree/Tree.stories.tsx [hunk 2]
```diff
@@ -54,84 +21,30 @@ export default {
...
-                (index > movedItemTargetIndex &&
-                    item.sort !== null &&
-                    previousItem &&
-                    previousItem.sort &&
-                    item.sort < previousItem.sort); // Following item with a sort prop lower than on previousItem
-
-            if (shouldUpdate) {
-                const incrementedPreviousItemSort = previousItem && previousItem.sort ? previousItem.sort + 1 : null;
-                const updatedSortValue = index === 0 ? 0 : incrementedPreviousItemSort;
+    const [nodesState, setNodes] = useState(apiNodes);
```
Comment (GoranRibic) [inline src/components/Tree/Tree.stories.tsx:24]: rename nodesState to nodes or setNodes to SetNodesState. Getter/Setter should have same name.

File: src/components/Tree/Tree.stories.tsx [hunk 3]
```diff
@@ -54,84 +21,30 @@ export default {
...
 
-                modifiedItem = { ...item, sort: updatedSortValue };
-                changedItemsIds.push(item.id);
+    const handleMove = (modifiedItems: DraggableItem<TreeListItem>[]) => {
```
Comment (GoranRibic) [inline src/components/Tree/Tree.stories.tsx:26]: add return type for handleMove

File: src/components/Tree/Tree.stories.tsx [hunk 4]
```diff
@@ -54,84 +21,30 @@ export default {
...
-
-        setNodes(listToTree(nodesClone));
-        apiNodes = nodesClone;
+        setNodes(modifiedArray);
     };
 
     return (
         <div style={{ maxWidth: "800px" }}>
-            <TreeComponent {...args} nodes={nodesState} onDrop={reorderItem} />
+            <TreeComponent {...args} nodes={nodesState} handleUpdate={handleMove} />
```
Comment (GoranRibic) [inline src/components/Tree/Tree.stories.tsx:41]: rename handleUpdate prop to onUpdate - props should have **on** prefix but actual function that handles that prop should have **handle** prefix

File: src/components/Tree/Node.tsx [hunk 5]
```diff
@@ -1,20 +1,22 @@
...
-import { IconProps } from "@foundation/Icon/IconProps";
 import { IconSize } from "@foundation/Icon/IconSize";
 import IconCaretDown from "@foundation/Icon/Generated/IconCaretDown";
 import IconCaretRight from "@foundation/Icon/Generated/IconCaretRight";
 import { merge } from "@utilities/merge";
 import { AnimatePresence, motion } from "framer-motion";
 import { useDrag } from "react-dnd";
-import { DropZone, DropZonePosition, OnDropCallback } from "@components/Tree/DropZone";
+import { DropZone, OnDropCallback } from "@components/DropZone";
+import { TreeListItem } from "@components/Tree/Tree";
```
[the remark this answers is not in the document]
Comment (Kenny806) [reply src/components/Tree/Node.tsx:11]: Done

File: src/components/Tree/Node.tsx [hunk 6]
```diff
@@ -27,27 +29,23 @@ export const renderNodeArray = (
             isFirst={i === 0}
             onDrop={onDrop}
             parentIds={parentIds}
+            listId={listId}
```
[the remark this answers is not in the document]
Comment (Kenny806) [reply src/components/Tree/Node.tsx:32]: I got either `treeId` (as `listId` comes from OrderableList) or something along the lines of `acceptedDnDItemType`. Suggestions are welcome

File: src/components/Tree/Tree.stories.tsx (hunk 2 above)
Comment (Kenny806) [reply src/components/Tree/Tree.stories.tsx:24]: Done

File: src/components/Tree/Tree.stories.tsx (hunk 3 above)
Comment (Kenny806) [reply src/components/Tree/Tree.stories.tsx:26]: Done

File: src/components/Tree/Tree.stories.tsx (hunk 4 above)
Comment (Kenny806) [reply src/components/Tree/Tree.stories.tsx:41]: Done

File: src/components/Tree/Node.tsx [hunk 7]
```diff
@@ -27,27 +29,23 @@ export const renderNodeArray = (
...
-    icon?: ReactElement<IconProps>;
-    label?: string;
-    value?: string;
-    nodes?: TreeNodeProps[];
-    actions?: React.ReactNode[];
+export interface TreeNodeProps extends TreeListItem {
+    nodes?: DraggableItem<TreeNodeProps>[];
 }
 
 type NodeProps = {
```
[the remark this answers is not in the document]
Comment (Kenny806) [reply src/components/Tree/Node.tsx:40]: Done

File: src/components/Tree/Node.tsx (hunk 1 above)
Comment (Kenny806) [reply src/components/Tree/Node.tsx]: Done

File: src/components/DropZone/DropZone.tsx [hunk 8]
```diff
@@ -50,8 +50,14 @@ export const DropZone = <T extends object>({ data, onDrop, children, listId }: D
             aria-hidden={!isActive}
             data-test-id="drop-zone"
             className={merge([
-                "tw-w-full tw-outline-none tw-relative tw-z-20 tw-h-[10px] tw-my-[-4px] tw-py-1",
-                isActive ? "tw-bg-violet-60 tw-bg-clip-content" : "",
+                "tw-w-full tw-transition-height",
+                data.position !== DropZonePosition.Within
```
[the remark this answers is not in the document]
Comment (Kenny806) [reply src/components/DropZone/DropZone.tsx]: Done

File: src/components/Tree/Node.tsx (hunk 1 above)
Comment (Kenny806) [reply src/components/Tree/Node.tsx]: Done

**Review by GoranRibic — APPROVED** [review]

**Pull request merged**